docs/by-example/sqlalchemy/advanced-querying.rst

```diff
@@ -0,0 +1,267 @@
+.. _sqlalchemy-advanced-querying:
+
+=============================
+SQLAlchemy: Advanced querying
+=============================
+
+This section of the documentation demonstrates running queries with
+aggregations, and queries using a fulltext index with analyzer, both
+using the CrateDB SQLAlchemy dialect.
+
+
+.. rubric:: Table of Contents
+
+.. contents::
+   :local:
+
+
+Introduction
+============
+
+Import the relevant symbols:
+
+    >>> import sqlalchemy as sa
+    >>> from sqlalchemy.ext.declarative import declarative_base
+    >>> from sqlalchemy.orm import sessionmaker
+    >>> from uuid import uuid4
+
+Establish a connection to the database, see also :ref:`sa:engines_toplevel`
+and :ref:`connect`:
+
+    >>> engine = sa.create_engine(f"crate://{crate_host}")
+    >>> connection = engine.connect()
+
+Create an SQLAlchemy :doc:`Session <sa:orm/session_basics>`:
+
+    >>> session = sessionmaker(bind=engine)()
+    >>> Base = declarative_base()
+
+
+Introduction to fulltext indexes
+================================
+
+:ref:`crate-reference:fulltext-indices` take the contents of one or more fields
+and split it up into tokens that are used for fulltext-search. The transformation
+from a text to separate tokens is done by an analyzer. In order to create
+fulltext search queries a :ref:`fulltext index with an analyzer
+<crate-reference:sql_ddl_index_fulltext>` must be defined for the related columns.
```

Comment:
Could restructure this a bit to make the flow more natural. For example:

```suggestion
:ref:`crate-reference:fulltext-indices` take the contents of one or more fields
and split it up into tokens that are used for fulltext-search. The
transformation from a text to separate tokens is done by an analyzer. In order
to create fulltext search queries we need to create a table with a
:ref:`fulltext index with an analyzer
<crate-reference:sql_ddl_index_fulltext>`::

    CREATE TABLE characters (
        id STRING PRIMARY KEY,
        name STRING,
        quote STRING,
        INDEX name_ft USING fulltext(name) WITH (analyzer = 'english'),
        INDEX quote_ft USING fulltext(quote) WITH (analyzer = 'english')
    )


We have to create this table using SQL because it is currently not possible to create ``INDEX`` fields using SQLAlchemy's declarative mapping. But we can define the table to use all other operations:


    >>> def gen_key():
    ...     return str(uuid4())

    >>> class Character(Base):
    ...     __tablename__ = 'characters'
    ...     id = sa.Column(sa.String, primary_key=True, default=gen_key)
    ...     name = sa.Column(sa.String)
    ...     quote = sa.Column(sa.String)
    ...     name_ft = sa.Column(sa.String)
    ...     quote_ft = sa.Column(sa.String)
    ...     __mapper_args__ = {
    ...         'exclude_properties': ['name_ft', 'quote_ft']
    ...     }


We define ``name_ft`` and ``quote_ft`` as regular columns but add them under ``__mapper_args__.exclude_properties`` to ensure they're excluded from insert or update operations.
```